core/src/main/java/org/radargun/stages/test/legacy/LegacyTestStage.java

```diff
@@ -169,7 +169,18 @@ public void run() {
             throw new IllegalStateException("Interrupted during ramp-up.", e);
          }
       }
-      started = true;
+      
+      //Wait until all threads are started to start stage
+      while (!isStarted()) {
+         int stressorStartedCount = 0;
+         for (LegacyStressor stressor : stressors) {
+            if (stressor.isStarted()) {
```

Comment:
Ouch, busy wait? `LegacyStressor.started` isn't volatile. Why don't you just use `CountDownLatch`?

Btw. the field is not volatile for a purpose, because it's repeatedly read in `recoding()` and given that it was set only once, it's better to keep the result in local field; otherwise we could use stage.isStarted() all the time.